### PR TITLE
BUG: DataFrame index name no longer resets after appending a list of series

### LINKED ISSUE #44109: DataFrame index name reset after appending a list of Series
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
pd.__version__
'1.4.0.dev0+938.g1bcb84175c'

df = pd.DataFrame([[1,2],[3,4]],index = pd.Index(['a','b'],name='keepthisname'))
              0  1
keepthisname      
a             1  2
b             3  4

serc = pd.Series([5,6],name='c')
0    5
1    6
Name: c, dtype: int64

dfc = df.append(serc)  # one at a time works
              0  1
keepthisname      
a             1  2
b             3  4
c             5  6
serd = pd.Series([7,8],name='d')  # as further evidenced with this...
dfc.append(serd)
              0  1
keepthisname      
a             1  2
b             3  4
c             5  6
d             7  8

df.append([serc,serd])  # but this resets the name of the index
   0  1
a  1  2
b  3  4
c  5  6
d  7  8
```

### Issue Description

DataFrame append of a list of Series conforming to the index resets the name of the index.

### Expected Behavior

The same as adding each element of the Series in a loop.

### Installed Versions

Built from source:

pd.__version__
'1.4.0.dev0+938.g1bcb84175c'

### PR DESCRIPTION
**Pull request by robbertjan94**
- [x] closes #44109
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Review by jbrockmendel — APPROVED** [review]
LGTM

File: pandas/tests/frame/methods/test_append.py [hunk 1]
```diff
@@ -96,6 +96,18 @@ def test_append_list_of_series_dicts(self):
...
 
+    def test_append_list_retain_index_name(self):
+        df = DataFrame(
+            [[1, 2], [3, 4]], index=pd.Index(["a", "b"], name="keepthisname")
+        )
+
+        serc = Series([5, 6], name="c")
+
+        result = df.append([serc])
+        expected = df.append(serc)
```
Comment (jreback) [inline pandas/tests/frame/methods/test_append.py:107]: can you explicitly construct the expected here. also would parameterize this test on both `[serc]` and `serc` in the append.

File: pandas/tests/frame/methods/test_append.py [hunk 2]
```diff
@@ -96,6 +96,18 @@ def test_append_list_of_series_dicts(self):
...
+        tm.assert_index_equal(result.index, expected.index)
```
Comment (jreback) [inline pandas/tests/frame/methods/test_append.py:108]: this is not needed (as L109 checks this)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/methods/test_append.py (hunk 1 above)
Comment (robbertjan94) [reply pandas/tests/frame/methods/test_append.py:107]: Hi @jreback, Thanks for your feedback. I guess that with your second sentence you mean to perform assert_frame_equal for append on `[serc]` and `serc` separetely?

Comment (jreback) [reply pandas/tests/frame/methods/test_append.py:107]: yep exactly

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @robbertjan94 very nice!

**Pull request merged**